hack/e2e.sh

```diff
@@ -95,79 +95,53 @@ function fail_test() {
+    if ! retry kube_event_exists "${namespace}" \
          "navigator-controller:ElasticsearchCluster:Normal:SuccessSync"
     then
         fail_test "Navigator controller failed to create SuccessSync event"
     fi
     if ! retry TIMEOUT=300 stdout_gt 0 kubectl \
-         --namespace "${USER_NAMESPACE}" \
+         --namespace "${namespace}" \
          get pilot \
          "es-test-mixed-0" \
          "-o=go-template={{.status.elasticsearch.documents}}"
     then
         fail_test "Elasticsearch pilot did not update the document count"
     fi
-    if ! kubectl delete \
-            --namespace "${USER_NAMESPACE}" \
-            ElasticSearchClusters \
-            --all; then
-        fail_test "Failed to delete elasticsearchcluster"
-    fi
-    if ! kube_delete_namespace_and_wait "${USER_NAMESPACE}"; then
-        fail_test "Failed to delete test namespace"
-    fi
-}
-
-test_elasticsearchcluster
-
-function ignore_expected_controller_errors() {
-    # Ignore the following error types:
-    # E1103 14:58:06.819858       1 reflector.go:205] github.com/jetstack/navigator/pkg/client/informers/externalversions/factory.go:68: Failed to list *v1alpha1.Pilot: the server could not find the requested resource (get pilots.navigator.jetstack.io)
-    # E1108 14:18:37.610718       1 reflector.go:205] github.com/jetstack/navigator/pkg/client/informers/externalversions/factory.go:68: Failed to list *v1alpha1.Pilot: an error on the server ("Error: 'dial tcp 10.0.0.233:443: getsockopt: connection refused'\nTrying to reach: 'https://10.0.0.233:443/apis/navigator.jetstack.io/v1alpha1/pilots?resourceVersion=0'") has prevented the request from succeeding (get pilots.navigator.jetstack.io)
-    egrep --invert-match \
-          -e 'Failed to list \*v1alpha1\.\w+:\s+the server could not find the requested resource\s+\(get \w+\.navigator\.jetstack\.io\)$' \
-          -e 'Failed to list \*v1alpha1\.\w+:\s+an error on the server \([^)]+\) has prevented the request from succeeding\s+\(get \w+\.navigator\.jetstack\.io\)$'
 }
 
-function test_logged_errors() {
-    if kubectl logs -c controller -l app=navigator,component=controller \
-            | egrep '^E[0-9]{4} ' \
-            | ignore_expected_controller_errors
-    then
-        fail_test "Unexpected errors in controller logs"
-    fi
-    if kubectl logs -c apiserver -l app=navigator,component=apiserver \
-            | egrep '^E[0-9]{4} '
-    then
-        fail_test "Unexpected errors in apiserver logs"
-    fi
-}
-
-test_logged_errors
-
-kubectl api-versions
-kubectl get apiservice -o yaml
-kubectl cluster-info dump --all-namespaces || true
+test_elasticsearchcluster "${USER_NAMESPACE}"
+if [ "${FAILURE_COUNT}" -gt "0" ]; then
+    fail_and_exit "${USER_NAMESPACE}"
+fi
 
-exit $FAILURE_COUNT
+if ! kubectl delete \
+        --namespace "${USER_NAMESPACE}" \
+        ElasticSearchClusters \
+        --all; then
+    fail_test "Failed to delete elasticsearchcluster" "${USER_NAMESPACE}"
+fi
```

Comment:
This was originally part of the E2E tests....to ensure that the API server allowed our resources to be deleted.
But that's now covered by the deletion of the namespace.
I'd be inclined to add `ElasticSearchClusters` to the list of resources that we explicitly delete in `kube_delete_namespace_and_wait`.